modules/nixos/tx5-signal-server.nix

```diff
@@ -0,0 +1,119 @@
+{
+  self,
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  cfg = config.services.tx5-signal-server;
+in {
+  options.services.tx5-signal-server = {
+    enable = lib.mkEnableOption "tx5-signal-server";
+
+    package = lib.mkOption {
+      default = self.packages.${pkgs.system}.tx5-signal-srv;
+      type = lib.types.package;
+    };
+
+    address = lib.mkOption {
+      description = "address to bind";
+      type = lib.types.str;
+    };
+
+    tls-port = lib.mkOption {
+      description = "port to bind for incoming TLS connections";
+      type = lib.types.int;
+    };
+
+    url = lib.mkOption {
+      description = "url for incoming TLS connections to the signal server";
+      type = lib.types.str;
+    };
+
+    port = lib.mkOption {
+      description = "port to bind";
+      type = lib.types.int;
+    };
+
+    iceServers = lib.mkOption {
+      description = "webrtc configuration to broadcast";
+      type = lib.types.listOf lib.types.attrs;
+      default = [];
+    };
+
+    demo = lib.mkEnableOption "enable demo broadcasting as a stand-in for bootstrapping";
+
+    configTextFile = lib.mkOption {
+      default = builtins.toFile "tx5.config.json" (builtins.toJSON {
```

Comment:
Nice, this is the kind of thing that I was most bothered by being missing previously